### PR TITLE
Add annotations when creating load balancer

### PR DESCRIPTION
**Pull request by yaocw2020**
Add annotations 
- loadbalancer.harvesterhci.io/network
- loadbalancer.harvesterhci.io/project
- loadbalancer.harvesterhci.io/namespace

IP pool module will select the IP pool automatically according to those annotations.

Related issue: https://github.com/harvester/harvester/issues/2134

Dependency chart PR: https://github.com/harvester/charts/pull/178

### CODE REVIEW

File: pkg/webhook/loadbalancer/mutator.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	var network string
+	if lb.Spec.WorkloadType == lbv1.Cluster && lb.Annotations != nil && lb.Annotations[utils.AnnotationKeyCluster] != "" {
+		network, err = m.findNetwork(lb.Namespace, lb.Annotations[utils.AnnotationKeyCluster])
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	annotations := lb.Annotations
+	if len(annotations) == 0 {
```
Comment (futuretea) [inline pkg/webhook/loadbalancer/mutator.go:56]: if annotations == nil

File: pkg/webhook/loadbalancer/mutator.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+	return project.Spec.DisplayName, nil
+}
+
+// Find the first network where the guest cluster is running
+// We assume that all the virtual machines composed the guest cluster are running in the same namespace and
+// have the same network configuration.
+func (m *mutator) findNetwork(namespace, clusterName string) (string, error) {
+	// list all the vmi instance in the same namespace of the load balancer
+	vmis, err := m.vmiCache.List(namespace, labels.Everything())
```
Comment (futuretea) [inline pkg/webhook/loadbalancer/mutator.go:115]: we can only list vmis created by node driver by using label `harvesterhci.io/creator=docker-machine-driver-harvester`

File: pkg/controller/loadbalancer/controller.go [hunk 3]
```diff
@@ -37,11 +38,13 @@ type Handler struct {
...
 	nadCache            ctlcniv1.NetworkAttachmentDefinitionCache
-	serviceClient       ctlCorev1.ServiceClient
-	serviceCache        ctlCorev1.ServiceCache
+	serviceClient       ctlcorev1.ServiceClient
+	serviceCache        ctlcorev1.ServiceCache
 	endpointSliceClient ctldiscoveryv1.EndpointSliceClient
 	endpointSliceCache  ctldiscoveryv1.EndpointSliceCache
 	vmiCache            ctlkubevirtv1.VirtualMachineInstanceCache
+	namespaceCache      ctlcorev1.NamespaceCache
+	projectCache        ctlmgmtcattle.ProjectCache
```
Comment (chrisho) [inline pkg/controller/loadbalancer/controller.go:47]: remove If unused

**Review by futuretea — DISMISSED** [review]
LGTM, thanks.

**Review by bk201 — DISMISSED** [review]
LGTM, thanks

**Review by bk201 — APPROVED** [review]

**Review by chrisho — APPROVED** [review]
LGTM, and it looks to work fine, thanks.
image

**Review by guangbochen — APPROVED** [review]
LGTM, verified that the guest LB with IP pool config can get IP correctly, thanks. 
```
apiVersion: loadbalancer.harvesterhci.io/v1beta1                                                                                                                                                     │
│ kind: LoadBalancer                                                                                                                                                                                   │
│ metadata:                                                                                                                                                                                            │
│   annotations:                                                                                                                                                                                       │
│     loadbalancer.harvesterhci.io/cluster: rke2-lb                                                                                                                                                    │
│     loadbalancer.harvesterhci.io/namespace: default                                                                                                                                                  │
│     loadbalancer.harvesterhci.io/network: default/vlan1238                                                                                                                                           │
│     loadbalancer.harvesterhci.io/project: c-m-28b8nfb4/p-srzp5                                                                                                                                       │
│   creationTimestamp: "2023-06-15T10:24:30Z"                                                                                                                                                          │
│   finalizers:                                                                                                                                                                                        │
│   - wrangler.cattle.io/harvester-lb-controller                                                                                                                                                       │
│   generation: 2                                                                                                                                                                                      │
│   labels:                                                                                                                                                                                            │
│     cloudprovider.harvesterhci.io/cluster: rke2-lb                                                                                                                                                   │
│     cloudprovider.harvesterhci.io/serviceName: foo                                                                                                                                                   │
│     cloudprovider.harvesterhci.io/serviceNamespace: default                                                                                                                                          │
│   name: rke2-lb-default-foo-a794f73b                                                                                                                                                                 │
│   namespace: default                                                                                                                                                                                 │
│   resourceVersion: "17018118"                                                                                                                                                                        │
│   uid: eb377caa-e3b8-4f0c-b776-f7b4a5408290                                                                                                                                                          │
│ spec:                                                                                                                                                                                                │
│   ipam: pool                                                                                                                                                                                         │
│   workloadType: cluster                                                                                                                                                                              │
│ status:                                                                                                                                                                                              │
│   address: 10.84.103.193                                                                                                                                                                             │
│   allocatedAddress:                                                                                                                                                                                  │
│     gateway: 10.84.103.254                                                                                                                                                                           │
│     ip: 10.84.103.193                                                                                                                                                                                │
│     ipPool: guest-rke2                                                                                                                                                                               │
│     mask: 255.255.255.192                                                                                                                                                                            │
│   conditions:                                                                                                                                                                                        │
│   - lastUpdateTime: "2023-06-15T10:24:30Z"                                                                                                                                                           │
│     status: "True"                                                                                                                                                                                   │
│     type: Ready                                                                                                                                                                                      │
```

**Pull request merged**